binding.c

```diff
@@ -6,8 +6,19 @@
 
 typedef struct {
   js_context_t *context;
+  bool is_destroyed;
 } bare_realm_t;
 
+static void
+on_realm_finalize (js_env_t *env, void *data, void *finalize_hint) {
+  bare_realm_t *realm = data;
+
+  if (realm->is_destroyed) return;
+
+  int err = js_destroy_context(env, realm->context);
+  if (err == 0) realm->is_destroyed = true;
```

Comment:
```suggestion
  assert(err == 0);

  realm->is_destroyed = true;
```

`js_destroy_context()` always returns 0, so this is safe to assert.